core/src/kvs/ds.rs

```diff
@@ -508,12 +509,52 @@ impl Datastore {
 
 	// Initialise the cluster and run bootstrap utilities
 	#[instrument(err, level = "trace", target = "surrealdb::core::kvs::ds", skip_all)]
-	pub async fn bootstrap(&self) -> Result<(), Error> {
-		// Insert this node in the cluster
-		self.insert_node(self.id).await?;
-		// Mark expired nodes as archived
-		self.expire_nodes().await?;
-		// Everything ok
+	pub async fn get_version(&self) -> Result<StorageVersion, Error> {
+		let tx = self.transaction(TransactionType::Write, LockType::Pessimistic).await?.enclose();
+
+		let version: StorageVersion =
+			match catch!(tx, tx.get(crate::key::storage::version::new(), None)) {
+				Some(v) => {
+					let bytes: [u8; 2] = v
+						.as_slice()
+						.to_owned()
+						.try_into()
+						.map_err(|_| Error::InvalidStorageVersion)?;
+
+					u16::from_be_bytes(bytes).into()
+				}
+				None => {
+					// If there are no keys in the database, this indicates that the database is new, and thus the latest storage version
+					let keys =
+						catch!(tx, tx.keys(crate::key::storage::version::suffix()..vec![0xff], 1));
+
+					let version = if keys.is_empty() {
+						StorageVersion::LATEST
+					} else {
+						StorageVersion::FIRST
+					};
+
+					// Persist the version in the datastore
+					let key = crate::key::storage::version::new();
+					let bytes = version.to_be_bytes().to_vec();
+					tx.set(key, bytes).await?;
+
+					version.into()
+				}
+			};
+
+		tx.commit().await?;
+
+		Ok(version)
+	}
+
+	// Initialise the cluster and run bootstrap utilities
+	#[instrument(err, level = "trace", target = "surrealdb::core::kvs::ds", skip_all)]
+	pub async fn set_version_latest(&self) -> Result<(), Error> {
+		let tx = self.transaction(TransactionType::Write, LockType::Pessimistic).await?;
+		let bytes = StorageVersion::LATEST.to_ne_bytes().to_vec();
```

Comment:
Did you not forget to update this @kearfy?

Reply:
Addressed in [58208b0](https://github.com/surrealdb/surrealdb/pull/4581/commits/58208b0a7a00612b525a96f2bf65122e429f023e)